field-value-counter-sink/src/main/java/org/springframework/cloud/stream/module/metrics/InMemoryFieldValueCounterRepository.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.module.metrics;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.springframework.util.Assert;
+
+/**
+ * Memory backed implementation of FieldValueCounterRepository that uses a ConcurrentMap
+ *
+ * @author Mark Pollack
+ * @author Ilayaperumal Gopinathan
+ *
+ */
+public class InMemoryFieldValueCounterRepository implements FieldValueCounterRepository {
+
+	private final ConcurrentMap<String, FieldValueCounter> map = new ConcurrentHashMap<>();
+
+	public FieldValueCounter save(FieldValueCounter fieldValueCounter) {
```

Comment:
these methods can be removed (or at minima made private) and their implementation inlined in `getOrCreate()`